plotly/tools.py

```diff
@@ -1451,6 +1455,823 @@ class FigureFactory(object):
+            # take a stat of the grouped data
+            stat = np.median(data_from_group)
+            # add to dictionary
+            group_stats[group] = stat
+
+        # create violin fig
+        fig = FF.create_violin(df, data_header='Score', group_header='Group',
+                               height=600, width=1000, use_colorscale=True,
+                              group_stats=group_stats)
+
+        # plot
+        py.iplot(fig, filename='Violin Plot with Colorscale')
+        ```
+        """
+        if _numpy_imported is False:
+            raise ImportError("FigureFactory.create_violin() requires "
+                              "numpy to be imported.")
+
+        if _scipy_imported is False:
+            raise ImportError("FigureFactory.create_violin() requires "
+                              "scipy to be imported.")
+
+        from plotly.graph_objs import graph_objs
+        from numbers import Number
+        plotly_scales = {'Greys': ['rgb(0,0,0)', 'rgb(255,255,255)'],
+                         'YlGnBu': ['rgb(8,29,88)', 'rgb(255,255,217)'],
+                         'Greens': ['rgb(0,68,27)', 'rgb(247,252,245)'],
+                         'YlOrRd': ['rgb(128,0,38)', 'rgb(255,255,204)'],
+                         'Bluered': ['rgb(0,0,255)', 'rgb(255,0,0)'],
+                         'RdBu': ['rgb(5,10,172)', 'rgb(178,10,28)'],
+                         'Reds': ['rgb(220,220,220)', 'rgb(178,10,28)'],
+                         'Blues': ['rgb(5,10,172)', 'rgb(220,220,220)'],
+                         'Picnic': ['rgb(0,0,255)', 'rgb(255,0,0)'],
+                         'Rainbow': ['rgb(150,0,90)', 'rgb(255,0,0)'],
+                         'Portland': ['rgb(12,51,131)', 'rgb(217,30,30)'],
+                         'Jet': ['rgb(0,0,131)', 'rgb(128,0,0)'],
+                         'Hot': ['rgb(0,0,0)', 'rgb(255,255,255)'],
+                         'Blackbody': ['rgb(0,0,0)', 'rgb(160,200,255)'],
+                         'Earth': ['rgb(0,0,130)', 'rgb(255,255,255)'],
+                         'Electric': ['rgb(0,0,0)', 'rgb(255,250,220)'],
+                         'Viridis': ['rgb(68,1,84)', 'rgb(253,231,37)']}
+
+        # Validate colors
+        if colors is None:
+            colors = DEFAULT_PLOTLY_COLORS
+
+        if isinstance(colors, str):
+            if colors in plotly_scales:
+                colors = plotly_scales[colors]
+
+            elif 'rgb' in colors:
+                # validate rgb color
+                colors = FigureFactory._unlabel_rgb(colors)
+                for value in colors:
+                    if value > 255.0:
+                        raise exceptions.PlotlyError("Whoops! The "
+                                                     "elements in your "
+                                                     "rgb colors "
+                                                     "tuples cannot "
+                                                     "exceed 255.0.")
```

Comment:
🐄 in general, your code ends up taking more lines than it needs to. Here, I'd suggest:

```
raise exception.PlotlyError(
    "Whoops! The elements in your rgb colors "
    "tuples cannot exceed 255.0."
)
```

It only saves a line in this case, but I think it also improves readability.